frontend/apps/reader/modules/readerzooming.lua

```diff
@@ -345,20 +402,63 @@ function ReaderZooming:genSetZoomModeCallBack(mode)
 end
 
 function ReaderZooming:setZoomMode(mode, no_warning)
-    if not no_warning and self.ui.view.page_scroll and self.paged_modes[mode] then
-        UIManager:show(InfoMessage:new{
-            text = T(_([[
+    mode = require("util").arrayContains(self.available_zoom_modes, mode) and mode or self.DEFAULT_ZOOM_MODE
+    if mode == "column" then
+        local zoom_factor = math.max(2, math.floor(self.zoom_factor))
+        self.zoom_factor = zoom_factor
+        self.zoom_pan_direction_vertical = true
+        self.zoom_pan_h_overlap = 0
+        self.ui:handleEvent(Event:new("ZoomPanUpdate", {
+            zoom_factor = zoom_factor,
+            zoom_pan_direction_vertical = true,
+            zoom_pan_h_overlap = 0,
+        }))
+    end
+    if not no_warning and self.ui.view.page_scroll then
+        local message
+        if self.paged_modes[mode] then
+            message = T(_([[
 %1
 
-In combination with continuous view (scroll mode), this can cause unexpected vertical shifts when turning pages.]]), self.paged_modes[mode]),
-            timeout = 5,
-        })
+In combination with continuous view (scroll mode), this can cause unexpected vertical shifts when turning pages.]]),
+                self.paged_modes[mode])
+        elseif self.panned_modes[mode] then
+            message = T(_([[
+%1
+
+You should enable it instead of continuous view (scroll mode).]]),
+                self.panned_modes[mode])
+        end
+        if message then
+            UIManager:show(InfoMessage:new{text = message, timeout = 5})
+        end
     end
 
     self.ui:handleEvent(Event:new("SetZoomMode", mode))
     self.ui:handleEvent(Event:new("InitScrollPageStates"))
 end
 
+function ReaderZooming:_zoomFactorChange(title_text)
+    UIManager:show(
+        SpinWidget:new{
+            width = math.floor(Screen:getWidth() * 0.6),
+            value = self.zoom_factor,
+            value_min = self.zoom_mode == "column" and 2 or 1.5,
+            value_max = 10,
+            value_step = self.zoom_mode == "column" and 1 or 0.1,
+            value_hold_step = 1,
+            precision = self.zoom_mode == "pan" and "%.1f" or nil,
```

Comment:
What happens when user sets `Pan zoom`, sets zoom to 1.5, and then selects `Column zoom`? Is that reset? What happens when user then selects `Column zoom`? Does it stay as `2`, or is it restored to `1.5`?

Reply:
`Pan zoom` -> `Column zoom` : `zoom_factor` becomes `2`, and `column mode` is enabled.
`Column zoom` -> `Pan zoom` : no change (`zoom_factor` stays `2` and `column mode` enabled).
The reason is not only to make things easier, but also that, after setting `Column zoom`, one may want to tune it more by the mean of `Pan zoom`.

Reply:
Alright, if it works as intended, then all good.